src/plugins/console/public/application/containers/editor/monaco/monaco_editor_actions_provider.ts

```diff
@@ -399,4 +399,66 @@ export class MonacoEditorActionsProvider {
       },
     ]);
   }
+
+  /**
+   * This function moves the cursor to the previous request edge (start/end line).
+   * If the cursor is inside a request, it is moved to the start line of this request.
+   * If there are no requests before the cursor, it is moved at the first line in the editor.
+   */
+  public async moveToPreviousRequestEdge() {
+    const currentPosition = this.editor.getPosition();
+    const model = this.editor.getModel();
+    if (!currentPosition || !model) {
+      return;
+    }
+    const { lineNumber: currentLineNumber } = currentPosition;
+    // Get all requests before the current line
+    const requestsBefore = await this.getRequestsBetweenLines(model, 1, currentLineNumber - 1);
+    if (requestsBefore.length === 0) {
+      // If no requests before current line, set position to first line
+      this.editor.setPosition({ lineNumber: 1, column: 1 });
+      return;
+    }
+    const lastRequestBefore = requestsBefore[requestsBefore.length - 1];
+    if (lastRequestBefore.endLineNumber < currentLineNumber) {
+      this.editor.setPosition({ lineNumber: lastRequestBefore.endLineNumber, column: 1 });
+    } else {
+      // If the end line of the request is after the current line, then the cursor is inside the request
+      // The previous request edge is the start line of the request
+      this.editor.setPosition({ lineNumber: lastRequestBefore.startLineNumber, column: 1 });
+    }
+  }
+
+  /**
+   * This function moves the cursor to the next request edge.
+   * If the cursor is inside a request, it is moved to the end line of this request.
+   * If there are no requests after the cursor, it is moved at the last line in the editor.
+   */
+  public async moveToNextRequestEdge() {
```

Comment:
I was thinking of adding tests for this function, but it turned out trickier than I expected. It uses many other complex functions that are hard to mock, especially `getRequestsBetweenLines`, which is a private function and I think it cannot be mocked. I'm open to any suggestions about testing.

Reply:
Update: I managed to add tests in [61e587b](https://github.com/elastic/kibana/pull/183504/commits/61e587b4b06efdac5e392b0b886dbfc300e5963e) by mocking some editor and model functions.